File: pinyougou-shop-web/src/main/resources/static/service/goodsService.js
// 商家管理 服务
app.service('goodsService', function ($http) {

    // 保存
    this.save = function (entity) {
        return $http.post('/consumer/goods', entity);
    };

    // 更新
    this.update = function (entity) {
        return $http({
            method: 'PUT',
            url: '/consumer/goods',
            data: entity
        });
    };

    // 分页条件查询
    this.findPage = function (currentPage, pageNum, entity) {
        return $http.post('/consumer/goods/page?currentPage=' + currentPage + '&pageNum=' + pageNum, entity);
    };

    // 详情
    this.findOne = function (id) {
        return $http.get('/consumer/goods/' + id);
    };

    // 批量删除
    this.deleteBatch = function (ids) {
        return $http({
            method: 'DELETE',
            url: '/consumer/goods/deleteBatch/' + ids
        })
    };

    // 上/下架
    this.updateMarketable = function (id, marketable) {
        return $http.get('/consumer/goods/updateMarketable/' + id + '/' + marketable);
    }

});

File: pinyougou-sellergoods-interface/src/main/java/com/xt/pinyougou/service/GoodsService.java
package com.xt.pinyougou.service;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.baomidou.mybatisplus.extension.service.IService;
import com.xt.pinyougou.entity.Goods;
import com.xt.pinyougou.entity.Item;
import com.xt.pinyougou.pojo.GoodsGroup;

import java.util.List;

/**
 * <p>
 *  商品 服务类
 * </p>
 *
 * @author xt
 * @since 2020-01-07
 */
public interface GoodsService extends IService<Goods> {
    boolean add(GoodsGroup goodsGroup);

    IPage<Goods> selectPage(Integer currentPage, Integer pageNum, Goods goods);

    GoodsGroup findOne(Long id);

    boolean updateGroup(GoodsGroup goodsGroup);

    boolean updateStatus(List<Long> ids, String status);

    boolean updateMarketable(Long id, String marketable);

    // 根据商品ID和状态查询Item表信息
    List<Item> findItemListByGoodsIdAndStatus(List<Long> goodsIds, String status);
}
